src/Build/Construction/Solution/SolutionProjectGenerator.cs

```diff
@@ -1013,6 +1013,7 @@ private ProjectInstance CreateTraversalInstance(string wrapperProjectToolsVersio
                 traversalProject,
                 _globalProperties,
                 explicitToolsVersionSpecified ? wrapperProjectToolsVersion : null,
+                _loggingService,
                 _solutionFile.VisualStudioVersion,
                 new ProjectCollection(),
```

Comment:
Was it not logging before because this private ProjectCollection didn't have a logger? It looks like ProjectInstance uses the ProjectCollection's logger by default. Should this thing inherit collection instead?

Reply:
Correct. I looked around for a collection at this point and didn't see one; I haven't traced it all the way back through to see if it's created at some point but not passed or if it's just created for the first time at a later point.

As I mentioned, I don't know what the significance of inheriting the ProjectCollection is versus just the LoggingService. If you think it's worth pushing harder, I'd assume several more hours to debug this, but I can.

Reply:
I think this is sufficient due diligence.